src/main/java/sg/ncl/MainController.java

```diff
@@ -2389,19 +2391,30 @@ public String saveExperimentImage(
             @PathVariable String expId,
             @PathVariable String nodeId) throws IOException {
 
-        if (saveImageForm.getImageName().length() < 2) {
+        //checking image name
+        String imageName = saveImageForm.getImageName();
+        if (imageName.length() < 2) {
             log.warn("Save image form has errors {}", saveImageForm);
-            redirectAttributes.addFlashAttribute("message", "Image name too short, minimum 2 characters");
+            redirectAttributes.addFlashAttribute("message", "Image name is required minimum of 2 characters");
+            return "redirect:/experiments/save_image/" + teamId + "/" + expId + "/" + nodeId;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "redirect:/experiments/save_image/" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)